src/modules/sensors/vehicle_optical_flow/VehicleOpticalFlow.cpp

```diff
@@ -215,7 +215,7 @@ void VehicleOpticalFlow::Run()
 			const float interval_us = 1e6f / _param_sens_flow_rate.get();
 
 			// don't allow publishing faster than SENS_FLOW_RATE
-			if (_integration_timespan_us < interval_us) {
+			if (sensor_optical_flow.timestamp_sample < _last_publication_timestamp + interval_us) {
```

Comment:
Can you keep both, but conditional on `sensor_optical_flow.integration_timespan_us == 0`?

Reply:
will do